---
title: "Data exploration of the immune response of wild mice against parasite infections"
author: "Fay"
date: "2024-02-01"
output: 
    pdf_document:
         keep_md: true
         fig_width: 12
         fig_height: 8
    html_document:
        toc: yes # table of content
        toc_float: 
            collapsed: false # show full toc
            smooth_scroll: true # toc scrolling behavior
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning = FALSE)
knitr::opts_chunk$set(fig.width=12, fig.height=8) 

```

```{r Libraries, include = FALSE, message = FALSE, warning=FALSE}
# Libraries
library(dplyr)
library(tidyr)
library(ggplot2)
library(ggridges)
library(knitr)
library(modelsummary)
library(janitor)
library(pheatmap)
library(tibble)
library(corrplot)
library(RColorBrewer)
library(leaflet)

```


```{r include = FALSE, message = FALSE, warning=FALSE}
# Read data
hm <- read.csv("Data/Data_output/imputed_clean_data.csv")

# divide into laboratory and field data set
lab <- hm %>% filter(origin == "Lab")
field <- hm %>% filter(origin == "Field")

###################### eimeria strains
# define the color for each level of infection
color_mapping <- c("E_falciformis" = "salmon", 
                   "E_ferrisi" = "forestgreen", 
                   "uninfected" = "blue")

## vectors for selecting columns
facs_lab <- c("CD4", "Treg", "Div_Treg", "Treg17", "Th1", "Div_Th1", "Th17", 
              "Div_Th17", "CD8", "Act_CD8", "Div_Act_CD8", "IFNy_CD4", 
              "IFNy_CD8") #"Treg_prop", "IL17A_CD4") 

facs_field <- c("CD4", "Treg", "CD4", "Treg17", "Th1", "Th17", "CD8",
                     "Act_CD8", "IFNy_CD4", "IL17A_CD4", "IFNy_CD8")

Genes_v   <- c("IFNy", "CXCR3", "IL.6", "IL.13", #"IL.10",
                "IL1RN","CASP1", "CXCL9", "IDO1", "IRGM1", "MPO", 
                "MUC2", "MUC5AC", "MYD88", "NCR1", "PRF1", "RETNLB", "SOCS1", 
                "TICAM1", "TNF") #"IL.12", "IRG6")
```

# Data structure of field and laboratory data sets
## Data originating from yearly field excursions in the House Mouse Hybrid Zone

GitHub repository: https://github.com/derele/Mouse_Eimeria_Field/tree/master

```{r include = FALSE, message = FALSE, warning=FALSE}
sota_org <- read.csv("https://raw.githubusercontent.com/derele/Mouse_Eimeria_Field/master/documentation/Documentation_MEF.csv")

kable(sota_org)
```

### Infection intensity data 
*Eimeria spp.* oocysts counting:

counter: character, Person who counted
Feces_g: numeric, amount of feces used in flotation
Date_count: date counted
N_oocysts_sq1 ... sq8: numeric, individual count for each single square on the neubaure chamber (up to 8)
mean_neubauer: numeric, mean of the 8 squares
PBS_dil_in_mL: numeric, in which volume of PBS whas
Ncells: number of neubauer "cells" (squares) counted
OPG: oocysts per gram feces (calculated from the above)
In some cases only OPG data might be available or rather data would need to be re-formatted to access all the raw values.

*Eimeria spp.* detection qPCR
We perform relative qPCR for detection and quantification of Eimeria DNA in intestinal tissue. We amplify a locus in the nuclear genome of the house mouse and a locus in the mitochondrial genome (COI) of Eimeria. We then calculate a "delta" between the two ct values.

delta_ct_ilwe_MminusE: threshold cycle for mouse minus Eimeria in Ileum tissue. Only E. vermiformis is (at low pervalence in Ileum tissue) and we therefore don't obtain this data-type for all years.

delta_ct_cewe_MminusE: threshold cycle for mouse minus Eimeria in Caecume tissue. E. ferrisi and E. falcifromis are detected here. We should have this (as coprehensively as possible) for every year!

MC.Eimeria: TRUE/FALSE. This was established in 2018 as an improvement over the '> -5 delta ct rule' for identification of Eimeria -positive qPCRs. Melting curves have to show a drastic drop at XXÂ°C to indicate melting of a proper Eimeria COI amplification product. It might be added where possible for per 2018 data post-hoc (if melting curves exist for a review of raw data).

## Data structure - laboratory challenge infections with *Eimeria spp.* 

Contains data for challenge (repeated) infections performed between 2017 and 2019.  The data product is structure in the following columns:

Mouse_ID: the unique identifier of the mouse
experiment: the experiment as numbered in the overview table
mouse_strain: the strain (inbred or outbred) of the mouse
primary_infection: The Eimeria strain used for the primary infection
challenge_infection: The Eimeria strain used for the challenge infection
infection_history: The resulting infection history
labels: the unique label of the fecal sample at a particular dpi
weight: the weight of the mouse at this dpi
weight_dpi0: the weight at the day of infection
relative_weight: the weight of the mouse at this dpi relative to the weight at dpi0
feces_weight: the weight of the feces collected at this dpi
dpi: days post infection at which samples and data in this row were taken
infection: the infection (primary or challenge) this row/dpi corresponds to
oocyst_sq1, oocyst_sq2, oocyst_sq3, oocyst_sq4: the raw values for squared during oocyst counting
dilution: the amount of PBS the feces (with it's relative weight) was dissolved in
OO4sq: the sum of oocysts in the four counting squares
OOC: the overall number of oocysts in in the feces (of a particular weight) at this dpi
OPG_O: Old way of counting opg (Emanuel ask me)
infection_type: what kind of infection are we looking at (challenge or primary, homologous or heterologous immunization). This is differently coded to infection, as here UNI:E88 (first uninfected, then infected with E88) would count as a "primaryE88" infection
The next values max_dpi until maximum_weight are calculated for the infection type in which the mice died

max_dpi = maximum dpi that the mouse that the mouse reached for each infection challenge or primary (group_by EH_ID and infection (primary/challenge) to get the value

maximum oocysts for each infection type (group_by EH_ID and infection (primary/challenge) to get the value)

maximum weight loss for each infection type (group_by EH_ID and infection (primary/challenge) to get the value)

death = challenge/primary (in which infection did the mouse die

Eim_MC = Melting curve for eimeria

delta = delta ct value


# Experimental planning - Laboratory infections
## Selected mouse strains 

* four wild-derived inbred mouse strains along with their respective F1 hybrids. 
* Two of these strains, SCHUNT and STRA, represent M. m. domesticus. 
* The strains BUSNA and PWD were derived from M. m. musculus
* Two intersubspecific hybrids (STRAxBUSNA and SCHUNTxPWD) and 
* Two intrasubspecific hybrids (SCHUNTxSTRA and PWDxBUSNA). 


```{r mouse_strains, warning=FALSE, echo = FALSE}
mouse_strains <- c("SCHUNT", "STRA", "BUSNA", "PWD", "STRA BUSNA", 
                   "SCHUNT PWD", "SCHUNT STRA", "PWD BUSNA")
hybrid_status <- c("M. m. domesticus", "M. m. domesticus", "M. m. musculus",
                   "M. m. musculus", "intersubspecific hybrids",
                   "intersubspecific hybrids", "intrasubspecific hybrids", 
                   "intrasubspecific hybrids")
strains <- as.data.frame(hybrid_status, mouse_strains)
kable(strains)
```

Numbers of each mouse strain

```{r strains,  warnings = FALSE, echo = FALSE}
lab %>%
    group_by(mouse_strain) %>%
    # Summarize the data to get counts for each mouse strain
    summarize(count = n()) %>%
    # Reorder mouse_strain by count
    mutate(mouse_strain = reorder(mouse_strain, count)) %>%
    # Plotting
    ggplot(aes(x = mouse_strain, y = count, fill = mouse_strain)) +
    geom_bar(stat = "identity") + 
    # Specify stat = "identity" for pre-summarized data
    geom_text(aes(label = count), vjust = -0.3) + 
    # Add count labels on top of bars
    scale_fill_viridis_d() + 
    # Use a nice color scale, like Viridis
    theme_minimal() + # Apply a minimal theme for a cleaner look
    theme(axis.text.x = element_text(angle = 50)) +
    labs(title = "Mouse strains in experimental infections", x = "Mouse Strain", 
         y = "Count") +# Add label
    guides(fill = "none") -> m_s

m_s

ggsave(filename = "~/GitHub/Article_predicting_WL/figures/mice_strains_n.jpeg",
        plot = m_s, width = 8, height = 6)
```


## Selected parasite strains

Infections were initiated by oral administration of 150 sporulated Eimeria oocysts
* Up to 16 species of Eimeria have been described from house mice 
* Overall prevalence in the wild 25.9% 
* Prevalence of *E. ferrisi* 14 %
* Prevalence of *E. falciformis* 4%

As a proxy for health we use the maximum relative weight lost during infection

Maximum weight loss = 
highest relative weight on any day of the experiment /
starting weight 

## Maximum relative weight loss in each infection group, in challenge and primary infections combined.


```{r general_WL_parasite, echo = FALSE, message=FALSE, warning=FALSE}
###################### eimeria strain
lab$current_infection <- gsub(pattern = "_", replacement = ". ", lab$current_infection)

ggplot(lab %>%
           filter(infection == "challenge"), 
       aes(x = WL_max, y = current_infection, fill = current_infection)) + 
    geom_density_ridges(jittered_points = TRUE, position = position_points_jitter(height = 0), 
                        scale = 0.9, alpha = 0.6, point_shape = 21, point_size = 2, point_alpha = 1) +
    geom_boxplot(width = 0.2, outlier.shape = NA, alpha = 0.5, position = position_nudge(x = 0.2)) +
   # coord_flip() +
    theme_minimal() +
    scale_color_manual(values = color_mapping) +
    theme(legend.position = "none",
          axis.text.x = element_text(angle = 0, vjust = 0.5, hjust=0.3)) +
    xlab("Maximum relative weight loss") +
    ylab("Parasite strain") -> eimeria_weight

eimeria_weight

```


Most of the mice in this experiment, are mice that have been challenged (infected for a second time). This replicates more accurately what happens in the wild. A much higher weight loss is expected in the primary infections.


```{r general_WL_parasite_infection, message=FALSE, echo = FALSE, warning=FALSE}
# prim vs challenge
lab$current_infection <- gsub(pattern = "_", replacement = ". ", lab$current_infection)

ggplot(lab, 
       aes(x = WL_max, y = current_infection, fill = current_infection)) + 
    geom_density_ridges(jittered_points = TRUE, position = position_points_jitter(height = 0), 
                        scale = 0.9, alpha = 0.6, point_shape = 21, point_size = 2, point_alpha = 1) +
    geom_boxplot(width = 0.2, outlier.shape = NA, alpha = 0.5, position = position_nudge(x = 0.2)) +
    # coord_flip() +
    theme_minimal() +
    theme(legend.position = "none",
          axis.text.x = element_text(angle = 0, vjust = 0.5, hjust=0.3)) +
    xlab("Maximum relative weight loss") +
    ylab("Parasite strain") +
    scale_color_manual(values = color_mapping) +
    facet_wrap(~ infection, nrow = 2)-> eimeria_weight_challenge

eimeria_weight_challenge
```


## Weight loss per mouse strain 

```{r mouse_strain_WL, message=FALSE, echo = FALSE, warning=FALSE}
# Define colors
colors <- c("TRUE" = "firebrick3", "FALSE" = "steelblue")

# transform mouse strain into factor
lab$mouse_strain <- as.factor(lab$mouse_strain)

lab$mouse_strain <- gsub(pattern = "_", " ", lab$mouse_strain)

# order factor levels
lab$mouse_strain <- factor(lab$mouse_strain, 
                           levels = names(
                               sort(tapply(lab$WL_max, lab$mouse_strain, median))))

ggplot(lab %>%
           filter(infection == "challenge"), 
       aes(x = WL_max, y = mouse_strain, fill = mouse_strain)) + 
    geom_density_ridges(jittered_points = TRUE, position = position_points_jitter(height = 0), 
                        scale = 0.9, alpha = 0.6, point_shape = 21, point_size = 2, point_alpha = 1) +
    geom_boxplot(width = 0.2, outlier.shape = NA, alpha = 0.5, position = position_nudge(x = 0.2)) +
    coord_flip() +
    theme_minimal() +
    theme(legend.position = "none",
          axis.text.x = element_text(angle = 70, vjust = 0.5, hjust=0.3)) +
    xlab("Maximum relative weight loss") +
    ylab("Mouse Strain") -> strains_weight

strains_weight

```



### Weight loss per mouse strain, challenge infections vs primary infections

```{r mouse_strain_WL_chal_prim, message=FALSE, echo = FALSE, warning=FALSE}

ggplot(lab, aes(x = WL_max, y = mouse_strain, fill = mouse_strain)) + 
    geom_density_ridges(jittered_points = TRUE, position = position_points_jitter(height = 0), 
                        scale = 0.9, alpha = 0.6, point_shape = 21, point_size = 2, point_alpha = 1) +
    geom_boxplot(width = 0.2, outlier.shape = NA, alpha = 0.5, position = position_nudge(x = 0.2)) +
    coord_flip() +
    theme_minimal() +
    theme(legend.position = "none",
          axis.text.x = element_text(angle = 70, vjust = 0.5, hjust=0.3)) +
    xlab("Maximum relative weight loss") +
    ylab("Mouse Strain") +
    facet_grid(~infection) -> strains_weight_challenge

strains_weight_challenge


```


## Preliminary data analysis immune data from laboratory infection experiments
### For how many mice do we have immune data? 
```{r , message=FALSE, warning=FALSE}
length(lab$Mouse_ID)
```

### How many mice in primary and how many in challenge infections?
```{r , message=FALSE, warning=FALSE}
lab %>%
    group_by(infection) %>%
    summarize(n())
```
### How many mice are there in each infection group?
```{r , message=FALSE, warning=FALSE}
lab %>%
    group_by(infection, current_infection) %>%
    summarize(n())
```

### For how many mice do we have FACS data? 
```{r , message=FALSE, warning=FALSE, echo = FALSE}
FACS_M <- lab %>%
    dplyr::select(Mouse_ID, all_of(facs_lab))

FACS <- FACS_M %>%
    dplyr::select(-Mouse_ID)

#remove rows with only nas
FACS <- FACS[,colSums(is.na(FACS))<nrow(FACS)]

#remove colums with only nas 
FACS <- FACS[rowSums(is.na(FACS)) != ncol(FACS), ]

FACS_M <- FACS_M %>%
    filter(row.names(FACS_M) %in% row.names(FACS))

length(FACS_M$Mouse_ID)
```

### For how many mice do we have immune gene expression data?
```{r , message=FALSE, warning=FALSE, echo = FALSE}
GENE_M <- lab %>%
    dplyr::select(Mouse_ID, all_of(Genes_v))

GENE <- GENE_M %>%
    dplyr::select(-Mouse_ID)

#remove rows with only nas
GENE <- GENE[,colSums(is.na(GENE))<nrow(GENE)]

#remove colums with only nas 
GENE <- GENE[rowSums(is.na(GENE)) != ncol(GENE), ]

GENE_M <- GENE_M %>%
    filter(row.names(GENE_M) %in% row.names(GENE))

length(GENE_M$Mouse_ID)
```

### How many mice have immune gene expression AND FACS data?
```{r, message=FALSE, warning=FALSE}
length(intersect(FACS_M$Mouse_ID, GENE_M$Mouse_ID))
```

For the complete FACS Data set, immune gene expression is complete too.


# Field infections - FACS immune cell data
### Number of mice with FACS data
```{r, message=FALSE, warning=FALSE, echo = FALSE}
FACS_M_f <- field %>%
    dplyr::select(Mouse_ID, all_of(facs_field))

FACS_f <- FACS_M_f %>%
    dplyr::select(-Mouse_ID)

#remove rows with only nas
FACS_f <- FACS_f[,colSums(is.na(FACS_f))<nrow(FACS_f)]

#remove colums with only nas 
FACS_f <- FACS_f[rowSums(is.na(FACS_f)) != ncol(FACS_f), ]

FACS_M_f <- FACS_M_f %>%
    filter(row.names(FACS_M_f) %in% row.names(FACS_f))

FACS_field <- field %>%
    filter(row.names(field) %in% row.names(FACS_f))

length(FACS_M_f$Mouse_ID)

```


### Number of mice with gene expression data 

```{r warning = FALSE, message = FALSE, echo = FALSE}
length(field$Mouse_ID)
```
### Capture locations of mice with gene expression data
```{r echo = FALSE, message = FALSE, warning = FALSE}
# Create a leaflet map
leaflet(data = field) %>%
  addTiles() %>%  # Add default OpenStreetMap tiles
  addCircleMarkers(~Longitude, ~Latitude, color = ~colorFactor(c("blue", "red"), HI)(HI),
                   popup = ~Mouse_ID) %>%
  addProviderTiles(providers$Stamen.Terrain) %>%
  setView(lng = mean(field$Longitude), lat = mean(field$Latitude), zoom = 7.5)

```

### Capture locations of mice with immune cell data

```{r echo = FALSE, message = FALSE, warning = FALSE}
# Create a leaflet map
leaflet(data = FACS_field) %>%
  addTiles() %>%  # Add default OpenStreetMap tiles
  addCircleMarkers(~Longitude, ~Latitude, color = ~colorFactor(c("blue", "red"), HI)(HI),
                   popup = ~Mouse_ID) %>%
  addProviderTiles(providers$Stamen.Terrain) %>%
  setView(lng = mean(field$Longitude), lat = mean(field$Latitude), zoom = 7.5)

```


## Immune cells
### Laboratory infections - heatmap of immune cells

```{r HEATMAP_lab_facs, message=FALSE, echo = FALSE, warning=FALSE, fig.width=10, fig.height=8}

annotation_df <- lab %>%
    filter(row.names(lab) %in% row.names(FACS)) %>%
    dplyr::select(c("Mouse_ID", "current_infection",
                    "mouse_strain", "WL_max")) 

lab_FACS <- lab %>%
    filter(row.names(lab) %in% row.names(FACS)) 
# turn the data frame into a matrix and transpose it. We want to have each cell 
# type as a row name 
FACS <- t(as.matrix(FACS_M))

#switch the matrix back to a data frame format
FACS <- as.data.frame(FACS)

# turn the first row into column names
FACS %>%
  row_to_names(row_number = 1) -> FACS

# Now further prepare the data frame for plotting by removing the first row
## and convert the column to row names with the cells 

heatmap_data <- FACS

# turn the columns to numeric other wise the heatmap function will not work
heatmap_data[] <- lapply(heatmap_data, function(x) as.numeric(as.character(x)))

annotation_df <- unique(annotation_df) %>%
    dplyr::filter(Mouse_ID %in% colnames(heatmap_data))

### Prepare the annotation columns for the heatmap
rownames(annotation_df) <- annotation_df$Mouse_ID

# Match the row names to the heatmap data frame
rownames(annotation_df) <- colnames(FACS)

#remove the unecessary column
annotation_df <- annotation_df %>% dplyr::select(-Mouse_ID, )

annotation_df$current_infection <- as.factor(annotation_df$current_infection)

heatmap_facs_LAB <- heatmap_data

color_mapping <- c("E. falciformis" = "salmon", 
                   "E. ferrisi" = "forestgreen", 
                   "uninfected" = "skyblue")

# Prepare the annotation_colors list to use in pheatmap
annotation_colors <- list(current_infection = color_mapping)


heatmap_data %>% 
  pheatmap(annotation_col = annotation_df,
           annotation_colors = annotation_colors,
           scale = "row")
```

Visual difference between infected and uninfected mice


### Field infections immune cells - heatmap
```{r HEATMAP_field_facs, message=FALSE, echo = FALSE, warning=FALSE, fig.width=10, fig.height=8}

annotation_df <- FACS_field %>% 
    dplyr::select(c("Mouse_ID", "MC.Eimeria",
                    "HI", "delta_ct_cewe_MminusE", all_of(facs_field))) %>%
    drop_na()


FACS <- annotation_df %>%
    dplyr::select(c(Mouse_ID, all_of(facs_field)))

annotation_df <- annotation_df %>%
    dplyr::select(-all_of(facs_field))

annotation_df$MC.Eimeria <- as.factor(annotation_df$MC.Eimeria)

# turn the data frame intFFACS_field# turn the data frame intFACS_field# turn the data frame into a matrix and transpose it. We want to have each cell 
# type as a row name 
FACS <- t(as.matrix(FACS))

#switch the matrix back to a data frame format
FACS <- as.data.frame(FACS)

# turn the first row into column names
FACS %>%
  row_to_names(row_number = 1) -> FACS

# Now further prepare the data frame for plotting by removing the first row
## and convert the column to row names with the cells 
heatmap_data <- FACS

# turn the columns to numeric other wise the heatmap function will not work
heatmap_data[] <- lapply(heatmap_data, function(x) as.numeric(as.character(x)))


annotation_df <- unique(annotation_df) %>%
    dplyr::filter(Mouse_ID %in% colnames(heatmap_data))

### Prepare the annotation columns for the heatmap
rownames(annotation_df) <- annotation_df$Mouse_ID

# Match the row names to the heatmap data frame
rownames(annotation_df) <- colnames(FACS)

#remove the unecessary column
annotation_df <- annotation_df %>% dplyr::select(-Mouse_ID, )


heatmap_data %>% 
  pheatmap(annotation_col = annotation_df,
           scale = "row")
```

Nothing to gain from this 


### Correlation between cells in laboratory challenge infections
```{r facs_core, echo = FALSE, warning=FALSE, message = FALSE}
FACS <- FACS_M %>%
    dplyr::select(-Mouse_ID)

# draw correlation between the cells
facs_correlation <- as.matrix(cor(FACS, 
                                  use="pairwise.complete.obs"))

# load the function to calculate the p value for correlations
source("~/GitHub/Article_predicting_WL/R/Functions/p_value_for_correlations.R")

# matrix of the p-value of the correlatio
p.mat <- cor.mtest(facs_correlation)

corrplot(facs_correlation, 
         method = "circle",  #method of the plot, "color" would show colour gradient
         tl.col = "black", tl.srt=45, #colour of labels and rotation
         col = brewer.pal(n = 8, name ="RdYlBu"), #colour of matrix
         order="hclust", #hclust reordering
         p.mat = p.mat, sig.level = 0.01, insig = "blank",
         addCoef.col = 'black',
         number.cex=0.5) 
  #Add significance level to the correlogram
#remove the values that are insignificant
```
### Correlations between cells in field mice

```{r facs_core_field, echo = FALSE, warning=FALSE, message = FALSE}
# draw correlation between the cells
facs_correlation <- as.matrix(cor(FACS_M_f %>%
                                      dplyr::select(-Mouse_ID), 
                                  use="pairwise.complete.obs"))

# matrix of the p-value of the correlatio
p.mat <- cor.mtest(facs_correlation)

corrplot(facs_correlation, 
         method = "circle",  #method of the plot, "color" would show colour gradient
         tl.col = "black", tl.srt=45, #colour of labels and rotation
         col = brewer.pal(n = 8, name ="RdYlBu"), #colour of matrix
         order="hclust", #hclust reordering
         p.mat = p.mat, sig.level = 0.01, insig = "blank",
         addCoef.col = 'black',
         number.cex=0.5) 
  #Add significance level to the correlogram
#remove the values that are insignificant
```


```{r immune_cells_differences_status, warning = FALSE,  echo = FALSE, message = FALSE}
f <- lab_FACS %>%
  pivot_longer(cols = all_of(facs_lab), names_to = "Cells", 
               values_to = "Percentage") 

f %>%
  ggplot(aes(x = current_infection, y = Percentage, 
             color = current_infection)) + 
  geom_boxplot() +
  geom_jitter() +
  facet_wrap(~ Cells) +
  labs( title = "Immune cell percentages in response to infection group, 
         laboratory infections") +
  theme_bw() +
  theme(axis.text.x=element_blank(),
        plot.title = element_text(size=30)) +
  scale_color_manual(values = color_mapping) 


```


- Missing: completing the data set of genotyping parasites in field samples 


